pyannote/database/custom.py

```diff
@@ -271,6 +293,87 @@ def subset_iter(
             {"uri": uri, "database": database, "subset": subset}, lazy=lazy_loader
         )
 
+def subset_trial(
+    self,
+    database: Text = None,
+    task: Text = None,
+    protocol: Text = None,
+    subset: Text = None,
+    entries: Dict = None,
+    database_yml: Path = None,
+):
+    """
+
+    Parameters
+    ----------
+    database : str
+        Database name (e.g. MyDatabase)
+    task : str
+        Task name (e.g. SpeakerDiarization, SpeakerVerification)
+    protocol : str
+        Protocol name (e.g. MyProtocol)
+    subset : {"train_trial", "development_trial", "test_trial"}
+        Subset
+    entries : dict
+        Subset entries.
+    """
+
+    if "trial" not in entries:
+        msg = f"Missing mandatory 'trial' entry in {database}.{task}.{protocol}.{subset}"
+        raise ValueError(msg)
+
+    lazy_loader = dict()
+
+    if 'duration' in entries.keys():
+        def try_with(file: ProtocolFile):
+            return Timeline(segments=[Segment(0, file['duration'])], uri=file['uri'])
+        lazy_loader['try_with'] = try_with
+        
+    elif 'annotated' in entries.keys():
+        def try_with(file: ProtocolFile):
+            return file['annotated']
+        lazy_loader['try_with'] = try_with
+
+    for key, value in entries.items():
+
+        if key == "uri" or key == "trial":
+            continue
+
+        if value.startswith("_"):
+            lazy_loader[key] = Template(value[1:], database_yml)
+
+        else:
+
+            path = resolve_path(Path(value), database_yml)
+
+            # check if file exists
+            if not path.is_file():
+                msg = f"No such file or directory: '{path}'"
+                raise FileNotFoundError(msg)
+
+            # check if loader exists
+            if path.suffix not in LOADERS:
+                msg = f"No loader for file with '{path.suffix}' suffix"
+                raise TypeError(msg)
+
+            # load custom loader class
+            Loader = LOADERS[path.suffix].load()
+            lazy_loader[key] = Loader(path)
```

Comment:
This is almost exactly the same code as the one used in `subset_iter`.
Can we factorize this in a `gather_loaders` function (we should probably find a better name) ?

Reply:
Done here : 971997d72dc9511e24ce75159ec70e1d5daafbc7
I used ```gather_loader``` for now... 
But the function is **collecting** every entry, **loading** the appropriate loader from the file and **storing** it in ```lazy_loader```, so gather seems ok...